spring-shell2-core/src/main/java/org/springframework/shell2/commands/Help.java

```diff
@@ -148,8 +157,14 @@ else if (description.keys().size() > 1) {
 			if (description.defaultValue().isPresent()) {
 				result
 						.append("  [Optional, default = ", AttributedStyle.BOLD)
-						.append(description.defaultValue().get(), AttributedStyle.BOLD.italic())
-						.append("]", AttributedStyle.BOLD);
+						.append(description.defaultValue().get(), AttributedStyle.BOLD.italic());
+				description.defaultValueWhenFlag().ifPresent(
+					s -> result.append(", or ", AttributedStyle.BOLD)
+						.append(s, AttributedStyle.BOLD.italic())
+					.append(" if used as a flag", AttributedStyle.BOLD)
+				);
+
+				result.append("]", AttributedStyle.BOLD);
 			} else {
 				result.append("  [Mandatory]", AttributedStyle.BOLD);
```

Comment:
Good point (even though that combination is arguably weird -- but still possible). Will fix